itertools is not included yet required when not using CLI feature

```
error[E0432]: unresolved import `itertools`
 --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/lune-0.8.2/src/lune/builtins/process/options/kind.rs:3:5
  |
3 | use itertools::Itertools;
  |     ^^^^^^^^^ use of undeclared crate or module `itertools`

error[E0599]: no method named `join` found for struct `std::iter::Map` in the current scope
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/lune-0.8.2/src/lune/builtins/process/options/kind.rs:58:26
   |
55 |                       ProcessSpawnOptionsStdioKind::all()
   |                       -----------------------------------
   |                       |
   |  _____________________method `join` is available on `&[kind::ProcessSpawnOptionsStdioKind]`
   | |
56 | |                         .iter()
57 | |                         .map(|k| format!("'{k}'"))
58 | |                         .join(", ")
   | |                         -^^^^ method not found in `Map<Iter<'_, ProcessSpawnOptionsStdioKind>, {closure@kind.rs:57:30}>`
   | |_________________________|
   | 
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,7 +22,6 @@ default = ["cli", "roblox"]
 cli = [
     "dep:anyhow",
     "dep:env_logger",
-    "dep:itertools",
     "dep:clap",
     "dep:include_dir",
     "dep:regex",
@@ -129,7 +128,7 @@ chrono_lc = "0.1"
 
 anyhow = { optional = true, version = "1.0" }
 env_logger = { optional = true, version = "0.11" }
-itertools = { optional = true, version = "0.12" }
+itertools = "0.12"
 clap = { optional = true, version = "4.1", features = ["derive"] }
 include_dir = { optional = true, version = "0.7", features = ["glob"] }
 regex = { optional = true, version = "1.7", default-features = false, features = [

